### PR TITLE
Fix the date format range.

### PR DESCRIPTION
**Pull request by GreatHumorist**
This is:

- [x] a bugfix

Checklist:

- [x] Changes are covered by unit tests
- [x] Code style is respected
- [x] Commit message explains **why** the change is made (see https://github.com/erlang/otp/wiki/Writing-good-commit-messages)
- [x] CHANGELOG.md contains a short summary of the change
- [ ] Documentation is updated as necessary

### Why this change is needed?
Fix the date format range in PhpOffice\PhpSpreadsheet\Shared::formattedPHPToExcel(), it should be in [1900/1/1, 9999/12/31], but the code before is not checked!
Now when a date is not in range give, the method will return false instead of break with wrong messages!

### CODE REVIEW

**Comment by GreatHumorist** [comment]
@PowerKiKi 
Hello, I have no idea that I can't change the CHANGELOG.md file to append change log, would you please help me to add it?

`- Fix the date format range, it should be in [1900/1/1, 9999/12/31], but the code before is not checked! - [#675](https://github.com/PHPOffice/PhpSpreadsheet/pull/675) [issue#669](https://github.com/PHPOffice/PhpSpreadsheet/issues/669)`

**Comment by GreatHumorist** [comment]
@gullij
Hello, this pr would help you to solve #669

File: src/PhpSpreadsheet/Shared/Date.php [hunk 1]
```diff
@@ -279,10 +279,15 @@ public static function timestampToExcel($dateValue)
...
      *
-     * @return float Excel date/time value
+     * @return false|float Excel date/time value
      */
     public static function formattedPHPToExcel($year, $month, $day, $hours = 0, $minutes = 0, $seconds = 0)
     {
+        // date range check, should in range [1900/1/1, 9999/12/31],
+        // the number is in range [0, 2958465]
+        if ($year < 1900 || $year > 9999) {
+            return false;
```
[the remark this answers is not in the document]
Comment (GreatHumorist) [reply src/PhpSpreadsheet/Shared/Date.php]: If there throws an Exception, some tests may not pass, like ./samples/Basic/20_Read_Excel2003XML.php.

[the remark this answers is not in the document]
Comment (PowerKiKi) [reply src/PhpSpreadsheet/Shared/Date.php]: I see there is a `1899-12-31T288:00:00.000` in `samples/templates/Excel2003XMLTest.xml` which is invalid according to the new rules in this PR.

I tested opening that file in MS Excel 2016 and it also errors and don't open. The MS Excel 2016 log file indeed contains an error about that invalid value:

```
XML ERROR in Table
REASON:	Bad Value
FILE:	C:\Users\Foo\AppData\Local\Microsoft\Windows\INetCache\Content.MSO\D5AC4727.xml
GROUP:	Cell
TAG:	Data
VALUE:	1899-12-31T288:00:00.000
```

So I think the file must be edited in order to not generate an error in Excel 2016 and our code should still throw an exception.

**Pull request closed without merging**